googlecloudstorage/googlecloudstorage.go

```diff
@@ -181,11 +207,34 @@ func parsePath(path string) (bucket, directory string, err error) {
 	return
 }
 
+func getServiceAccountClient(keyJsonfilePath string) (*http.Client, error) {
+	data, err := ioutil.ReadFile(os.ExpandEnv(keyJsonfilePath))
+	if err != nil {
+		return nil, fmt.Errorf("error opening credentials file: %v", err)
+	}
+	conf, err := google.JWTConfigFromJSON(data, storageConfig.Scopes...)
+	if err != nil {
+		return nil, fmt.Errorf("error processing credentials: %v", err)
+	}
+	return conf.Client(context.TODO()), nil
```

Comment:
Moved to the `oauth2.Context` magic of having an http client in the context. That stuff is convoluted ;)

Anyway, tested, works as before.